crates/contract/src/lib.rs

```diff
@@ -2950,4 +2962,90 @@ mod tests {
             ProtocolContractState::Initializing(gen_initializing_state(2, 1).1);
         test_proposed_updates_case_given_state(protocol_contract_state);
     }
+
+    #[test]
+    pub fn test_remove_update_vote_running() {
+        let running_state = gen_running_state(2);
+        let participants = running_state.parameters.participants().clone();
+        let protocol_contract_state = ProtocolContractState::Running(running_state);
+        let mut contract = MpcContract::new_from_protocol_sate(protocol_contract_state);
+        let expected = propose_and_vote_code(0, &mut contract);
+        for (account_id, _, _) in participants.participants() {
+            contract
+                .proposed_updates
+                .vote(&UpdateId(0), account_id.clone());
+            let mut expected_with_participant_vote = expected.clone();
+            expected_with_participant_vote
+                .votes
+                .push(account_id.into_dto_type());
+            expected_with_participant_vote.votes.sort();
+            let mut res = contract.proposed_updates();
+            res.0.iter_mut().for_each(|update| update.votes.sort());
+            assert_eq!(
+                res,
+                dtos::ProposedUpdates(vec![expected_with_participant_vote])
+            );
+
+            testing_env!(VMContextBuilder::new()
+                .signer_account_id(account_id.as_v1_account_id())
+                .predecessor_account_id(account_id.as_v1_account_id())
+                .build());
+
+            contract.remove_update_vote();
+            let mut expected_without_participant_vote = expected.clone();
+            expected_without_participant_vote.votes.sort();
+            let mut res = contract.proposed_updates();
+            res.0.iter_mut().for_each(|update| update.votes.sort());
+            assert_eq!(
+                res,
+                dtos::ProposedUpdates(vec![expected_without_participant_vote])
+            );
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "not a voter")]
+    fn test_remove_update_vote_panics_if_non_voter() {
+        let running_state = gen_running_state(2);
+        let protocol_contract_state = ProtocolContractState::Running(running_state);
+        let mut contract = MpcContract::new_from_protocol_sate(protocol_contract_state);
```

Comment:
not introduced in this PR, but if it does not break API, we should:
```suggestion
        let mut contract = MpcContract::new_from_protocol_state(protocol_contract_state);
```